### DISCUSSION TITLE
May I ask if the rain and fog in Carla will affect the LiDAR?

### DISCUSSION THREAD

**Discussion by BG888** [carla-simulator/carla#7673] [Q&A]
Recently, I have been using Carla to simulate sensor data in special scenarios, but it seems that rainy and foggy weather only affects the camera and not the LiDAR. Is that the case?

**Comment by xavisolesoft** [comment]
Not, LIDAR just detects objects that have a RayCast collider mesh.
Please, find mode information about it on the next links:
https://carla.readthedocs.io/en/0.9.12/tuto_A_add_vehicle/#:~:text=5.%20Create%20the%20mesh%20for%20the%20raycast%20sensor.
https://carla.readthedocs.io/en/0.9.11/tuto_D_generate_colliders/

**Reply by BG888** [reply]
Thank you for your reply, it's very useful！
